xclim/sdba/processing.py

```diff
@@ -152,15 +140,71 @@ def jitter_over_thresh(x: xr.DataArray, thresh: str, upper_bnd: str) -> xr.Datas
     -----
     If thresh is low, this will change the mean value of x.
     """
-    thresh = convert_units_to(thresh, x)
-    upper_bnd = convert_units_to(upper_bnd, x)
-    if uses_dask(x):
-        jitter = dsk.random.uniform(
-            low=thresh, high=upper_bnd, size=x.shape, chunks=x.chunks
-        )
-    else:
-        jitter = np.random.uniform(low=thresh, high=upper_bnd, size=x.shape)
-    out = x.where(~((x > thresh) & (x.notnull())), jitter.astype(x.dtype))
+    return jitter(x, lower=None, upper=thresh, minimum=None, maximum=upper_bnd)
+
+
+@update_xclim_history
+def jitter(
+    x: xr.DataArray,
+    lower: str = None,
+    upper: str = None,
+    minimum: str = None,
+    maximum: str = None,
+):
+    """Replaces values under a threshold and values above another by a uniform random noise.
+
+    Do not confuse with R's jitter, which adds uniform noise instead of replacing values.
+
+    Parameters
+    ----------
+    x : xr.DataArray
+      Values.
+    lower : str
+      Threshold under which to add uniform random noise to values, a quantity with units.
+      If None, no jittering is performed on the lower end.
+    upper : str
+      Threshold over which to add uniform random noise to values, a quantity with units.
+      If None, no jittering is performed on the upper end.
+    minimum : str
+      Lower limit (excluded) for the lower end random noise, a quantity with units.
+      If None but `lower` is not None, 0 is used.
+    maximum : str
```

Comment:
The naming is a bit confusing, because random.uniform uses low and high as parameters.

Reply:
What does  @RondeauG  think?
I reunited the two jitters in a single function, so it's easier to use with variables like `hurs` or `huss`.

Examples:
```python3
pr = jitter(pr, lower='0.01 mm/d', minimum='0 mm/d') # minimum defaults to that, I could have skipped it.
hurs = jitter(hurs, lower='0.01 %', upper='99.99 %', minimum='0 %', maximum='100 %')
```

Reply:
This looks fine to me, I like the idea of combining both functions. No strong opinion on `high/low` vs. `upper/lower`